grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -410,7 +410,8 @@ class UserService {
         def fullConfig = grailsApplication.config.openboxes.dashboardConfig
         def mainDashboardId = grailsApplication.config.openboxes.dashboardConfig.mainDashboardId
         def resultConfig = [
-                dashboard: fullConfig.dashboards[id ?: mainDashboardId],
+                // Clone to not overwrite the main config in the line 422
+                dashboard: fullConfig.dashboards[id ?: mainDashboardId].clone(),
```

Comment:
I like the idea, but how would we overwrite. grailsApplication.config is stored in memory and can be overwritten? Oh shit, is that why some users see other user's dashboards? 

If so, yikes. Good catch.

Reply:
@jmiranda Yeah I was confused by this too, but seems that this line `resultConfig.dashboard.personal = personalDashboard` overwrites it for at least "some" time (I am not sure when or if it's refreshed). Long enough to log in on one user and then after logging to another user, that has no personal dashboard modified, to be able to see the previous one's dashboard pulled from the **default** config.

Reply:
So it might be a case when there is loggin in/out from one user to another involved. I have not tried this for two different users in two separate sessions at the same time.